### PR TITLE
Fix #2141: Add feature for filtering the submissions

### LINKED ISSUE #2141: Add feature for filtering the submissions
## Current Scenario

If there are a large number of submissions in a challenge and a challenge host wants to view all the submissions for a participant team then they have to scroll through all the submissions.

## Expected

The challenge host should be able to filter the submissions by participant team name in `View All Submissions` Tab.

## Deliverables
- [ ] Add the required API's.
- [ ] Add the required frontend UI.

Please feel free to discuss the approach before start coding.

### PR DESCRIPTION
**Pull request by Sanji515**
**Deliverables**
- [x] Added the required API
- [x] Added the required frontend UI

There are three ways a user can filter the query:

1. First with the default when a user does not select any option from the dropdown, it will be filtered as `submission ID` or `participant team name`.
2. Second, when a user selects `Submission-ID` from the dropdown, the submission list will be filtered as `Submission-ID`.
3. And last when a user selects `Team Name` and will be filtered as `participant team name`. 

Here's a screencast of it-->

recording 4

### CODE REVIEW

**Comment by Sanji515** [comment]
@RishabhJain2018 @deshraj Please review.

**Comment by vkartik97** [comment]
Hi @Sanji515 !
Can you please add Screenshot in PR description?

**Comment by Sanji515** [comment]
@vkartik97 Should I add a screencast.

**Comment by RishabhJain2018** [comment]
@Sanji515 Please resolve the conflicts here.

**Comment by Sanji515** [comment]
> @Sanji515 Please resolve the conflicts here.

@RishabhJain2018 Done, please review

File: apps/challenges/views.py [hunk 1]
```diff
@@ -69,6 +69,8 @@
     get_participant_team_of_user_for_a_challenge,
 )
 
+from jobs.filters import SubmissionFilter
```
Comment (RishabhJain2018) [inline apps/challenges/views.py]: Please add this import alphabetically with other imports from `jobs` app. i.e. line 59

File: apps/challenges/views.py [hunk 2]
```diff
@@ -69,6 +69,8 @@
...
+
```
Comment (RishabhJain2018) [inline apps/challenges/views.py]: Please remove this line.

File: apps/challenges/views.py [hunk 3]
```diff
@@ -56,6 +56,7 @@
     is_user_a_host_of_challenge,
     get_challenge_host_team_model,
 )
+from jobs.filters import SubmissionFilter
```
Comment (RishabhJain2018) [inline apps/challenges/views.py]: Please import it alphabetically :P

**Pull request merged**